Add Multi-Profile Support (#1183)
Sage-Bionetworks/synapsePythonClient: docs/tutorials/authentication.md

```python
import synapseclient
syn = synapseclient.login(authToken="authtoken")

> #returns Welcome, First Last! 

Reviewer: nit: is this something meant to be deleted?

Reviewer 2: I think this is a note for what should be printed back to the user if they have succeeded in authenticating. This shows up in the mkdocs site and I think its pretty helpful.

Logged in as: username (1234567)
```

> ## Use `.synapseConfig` for Multiple Profiles

Reviewer 2: I think the documentation does a good job describing how to use the `.synapseConfig` file, but I think the title was fine as it was since we can assume that the basic use case will still be users with only one profile. If anything, we should have a subsection within this one that describes how to go from using one profile to using multiple.
```suggestion
## Use `.synapseConfig`
```

For writing code using the Synapse Python client that is easy to share with others, please do not include your credentials in the code. Instead, please use the `~/.synapseConfig` file to manage your credentials.

The Synapse Python Client supports multiple profiles within the ~/.synapseConfig file, enabling users to manage credentials for multiple accounts.

> Each profile is defined in its own '[profile <profile_name>]' section. A default profile can still be defined using '[default]'.

Reviewer: nit: I think it might make sense to just use backtick `[profile <profile_name>]`

Author: Thanks for the suggestion! I updated the formatting with backticks for clarity.

When installing the Synapse Python client, the `~/.synapseConfig` is added to your home directory.

### Automatically modifying the `~/.synapseConfig` file with the Command line Client
You may modify the `~/.synapseConfig` file by utilizing the [command line client command and following the interactive prompts](./command_line_client.md#config):

<!-- termynal -->
```

> #For Default profile

Reviewer: To make this more clear, maybe you could say something like:
"for modifying the username used in the default profile"

> synapse --profile default config
Synapse username (Optional): $MY_USERNAME

Auth token: $MY_SYNAPSE_TOKEN

> #For additional profiles

Reviewer: similarly, I think here you could say something like: "for adding a new profile with a new username"

> synapse --profile $MY_PROFILE_NAME config
Synapse username (Optional): $MY_USERNAME

> 

Reviewer: I tried adding a new profile using the command here, and I got message: 
```
[ERROR] Failed to perform command: <function config at 0x107d7edd0>
Traceback (most recent call last):
  File "/Users/lpeng/Documents/synapsePythonClient/venv/lib/python3.10/site-packages/synapseclient/__main__.py", line 1807, in perform_main
    args.func(args, syn)
  File "/usr/local/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/Users/lpeng/Documents/synapsePythonClient/venv/lib/python3.10/site-packages/synapseclient/__main__.py", line 591, in config
    with open(args.configPath, "w") as config_o:
PermissionError: [Errno 13] Permission denied: '/Users/lpeng/.synapseConfig'
```
Can we guide user in the tutorial so that this message can be avoided? I assume user needs to grant permission for .synapseConfig to be modified.

Author: Smart! Thank you!

Auth token: $MY_SYNAPSE_TOKEN
```

### Manually modifying the `~/.synapseConfig` file
The following describes how to add your credentials to the `~/.synapseConfig` file without the use of the `synapse config` command.

Open the `~/.synapseConfig` file using your preferred text editing tool and find the following section:

```
#[default]
#username = default_user
#authtoken = default_auth_token

#[profile user1]
#username = user1
#authtoken = user1_auth_token

#[profile user2]
#username = user2
#authtoken = user2_auth_token
```

To enable this section, uncomment it. You don't need to specify your username when using authtoken as a pair, but if you do, it will be used to verify your identity. A personal access token generated from your synapse.org Settings can be used as your .synapseConfig authtoken.
```
[default]
username = default_user

> #authtoken = default_auth_token

Reviewer 2: should this be uncommented too?

[profile user1]
username = user1
authtoken = user1_auth_token

[profile user2]
username = user2
authtoken = user2_auth_token
```

If logging in without specifying any profiles, it will default to the default profile:
Now, you can login without specifying any arguments:

```python
import synapseclient
syn = synapseclient.login()
```

> If logging in specifying the profile, it will log in with said profile:

Reviewer 2: ```suggestion
Or, if you are logging into a specific profile, simply pass the profile name as an argument to `login()`: 
```

The Synapse Python Client supports multiple profiles within the ~/.synapseConfig file, enabling users to manage credentials for multiple accounts.

Each profile is defined in its own '[profile <profile_name>]' section. A default profile can still be defined using '[default]'.

When installing the Synapse Python client, the `~/.synapseConfig` is added to your home directory.

### Automatically modifying the `~/.synapseConfig` file with the Command line Client
You may modify the `~/.synapseConfig` file by utilizing the [command line client command and following the interactive prompts](./command_line_client.md#config):

<!-- termynal -->
```
#For Default profile
> synapse --profile default config
Synapse username (Optional): $MY_USERNAME

Auth token: $MY_SYNAPSE_TOKEN

#For additional profiles
> synapse --profile $MY_PROFILE_NAME config
Synapse username (Optional): $MY_USERNAME

Auth token: $MY_SYNAPSE_TOKEN
```

### Manually modifying the `~/.synapseConfig` file
The following describes how to add your credentials to the `~/.synapseConfig` file without the use of the `synapse config` command.

> Open the `~/.synapseConfig` file (can use 'nano ~/.synapseConfig' to edit or 'cat ~/.synapseConfig' to read from the CLI)
> and find the following section:

Reviewer 3: ```suggestion
Open the `~/.synapseConfig` file using your preferred text editing tool and find the following section:
```
